### DISCUSSION TITLE
How the published app use custom domain?

### DISCUSSION THREAD

**Discussion by hjh1004** [Budibase/budibase#3969] [Help] [answered]
Hello. Budibase budi(friends)!
I want to each published app use custom domain.
For example, first published admin app(id: aaa) use "admin-a-mydomain.com" domain and
second published admin app(id: bbb) use "admin-b-mydomain.com" domain.
So I set up Route53 two domain and nginx configure below code in AWS EC2 after running budibase docker-compose.

```
server {
        set $budibase_host      127.0.0.1;
        set $budibase_port      10000;
        set $admin_a_domain admin-a-mydomain.com;
        set $first_app_id   aaa;

        listen 80;
        listen [::]:80;
        server_name $admin_a_domain;

        location / {
                proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
                proxy_set_header X-Forwarded-Proto $scheme;
                proxy_set_header X-Real-IP $remote_addr;
                proxy_set_header Host $http_host;
                proxy_http_version 1.1;
                proxy_pass http://$budibase_host:$budibase_port/$first_app_id;
        }
}

server {
        set $budibase_host      127.0.0.1;
        set $budibase_port      10000;
        set $admin_b_domain  admin-b-mydomain.com;
        set $second_app_id   bbb;

        listen 80;
        listen [::]:80;
        server_name $admin_b_domain;

        location / {
                proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
                proxy_set_header X-Forwarded-Proto $scheme;
                proxy_set_header X-Real-IP $remote_addr;
                proxy_set_header Host $http_host;
                proxy_http_version 1.1;
                proxy_pass http://$budibase_host:$budibase_port/$second_app_id;
        }
}
```

I expect that I connect "admin-a-mydomain.com" and then open first admin "aaa" app.
but If I connect "admin-a-mydomain.com", then open budibase portal(not published "aaa" app)
So can you describe how to use reverse proxy to use custom domain?
Is it impossible now excluding "mydomain.com/app/{app_name}"

reverse-proxy documentation can handle that only portal site use custom domain.
https://docs.budibase.com/self-hosting/hosting-settings/advanced-options/reverse-proxy

**Comment by mjashanks** [comment, marked as answer]
I have never seen anyone try this, and we don't support it by default - so I'm not really sure how to accomplish this, or if it is even possible.

One issue that you will come up against is login. Users login to the Budibase platform as a whole, not per app. i.e. login will only work with one domain. 

My sense is that this will not be possible.

**Reply by SantoshSrinivas79** [reply]
Ideally, I would spin up multiple instances of Budibase on different ports for completely different domains.
Also, limit one instance to one app

**Reply by hjh1004** [reply]
Thank you for a clear answer!

**Comment by burkeazbill** [comment]
I found this thread by searching Route53 as I was hoping someone had done some DNS automation using Budibase as a front-end for requests (my use-case)... but instead, this is the only thread that came up. I actually REALLY like this use case and think it would be a GREAT feature! Sure, you can spin up one budibase instance per desired FQDN, but overall that becomes more resource intensive. 

If a user is only entitled to a single app, I think it would be better if (after authentication) they were simply directed to the app rather than defaulting to the Apps page with that one app. While it wouldn't address the OP's specific use case, it would be helpful.
